libs/net/vmspec.py

```
@@ -194,3 +198,67 @@ def wait_for_ifaces_status(
+    if any(cond.type == condition and cond.status == status for cond in vmi_instance.status.conditions or []):
+        return
+
+    for event in vmi.watcher(timeout=timeout, resource_version=vmi_instance.metadata.resourceVersion):
+        if event["type"] != "MODIFIED":
+            continue
+        if any(cond.type == condition and cond.status == status for cond in event["object"].status.conditions or []):
+            return
+
+    raise VMConditionNotFoundError(
+        f"Condition {condition}={status} was not reached on VMI {vm.name} within {timeout}s."
+    )
```

Ruff (0.15.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)